tests/urlHash.js

```diff
@@ -29,19 +29,28 @@ requirejs([
             assert.equal(mug.absolutePath, "/data/group");
         });
 
-        it("should change the hash when you add a question", function() {
+        it("should change the hash when you add a question", function(done) {
             util.loadXML("");
             util.addQuestion("Text", "text");
-            assert.equal(window.location.hash, '#form/text');
+            util.saveAndReload(function(){
+                assert.equal(window.location.hash, '#form/text');
+                done();
+            })
```

Comment:
Why did these tests need to change? Seems like there might be a change in functionality. Is it now necessary to save and reload the form before the URL is updated with new hash?

Reply:
The place where we have added handlers to `setURLHash` is when a `question id` is modified or when a question is saved. But in the [tests utils, addQuestion method](https://github.com/dimagi/Vellum/blob/master/tests/utils.js#L479) does not modify the question id of the question so the trigger isn't pulled. But on FormBuilder the question id is modified, so the trigger is pulled.
So for the tests to work fine I had to save the questions.

Reply:
Are you saying that `util.addQuestion("Text", "text")` does not trigger the `question id` modified event? It seems better to trigger that event than using `util.saveAndReload` in these tests. What do you think?

It should be possible to trigger a question id change by changing the question id explicitly after creating the question. Something like
```js
            var text = util.addQuestion("Text", "initial");
            text.p.nodeID = "text";
            assert.equal(window.location.hash, '#form/text');
```

Reply:
Manually triggering the event seems much better. I will try if that works.

Reply:
Oh, right. Since this is testing functionality that is implemented in the UI layer (widgets.js), it should trigger the widget change event. Here's an example:
```js
var text = util.addQuestion("Text", "initial");
util.clickQuestion("initial");
$("[name=property-nodeID]").val("text").change();
assert.equal(window.location.hash, '#form/text');
```

Reply:
I realized that we can just update the hash by clicking on the question, so just `util.clickQuestion` worked out fine.